examples/pumi/ex6p.cpp

```diff
@@ -0,0 +1,381 @@
+//                       PUMI Example 6 - Parallel Version
```

Comment:
TODO: add PUMI examples section in `doc/CodeDocumentation.dox` (similar to SUNDIALS and PETSc)